core/chains/evm/txmgr/orm.go

```diff
@@ -42,6 +42,78 @@ type orm struct {
 
 var _ ORM = (*orm)(nil)
 
+// Structs saved in database should be internal to
+// the ORM and distinct from generic params
+// so that modifying generic params does neccessitate
+// modification of underlying column types
+// (or runtime errors by mistake)
+type dbReceipt struct {
+	ID               int64
+	TxHash           common.Hash
+	BlockHash        common.Hash
+	BlockNumber      int64
+	TransactionIndex uint
+	Receipt          evmtypes.Receipt
+	CreatedAt        time.Time
+}
+
+type dbRawReceipt = evmtypes.Receipt
+
+type dbReceiptPlus struct {
+	ID           uuid.UUID        `db:"id"`
+	Receipt      evmtypes.Receipt `db:"receipt"`
+	FailOnRevert bool             `db:"FailOnRevert"`
+}
+
+func fromDBReceipts(rs []dbReceipt) (receipts []EvmReceipt) {
+	for i := 0; i < len(rs); i++ {
+		r := rs[i]
+		receipts = append(receipts, EvmReceipt{
+			ID:               r.ID,
+			TxHash:           r.TxHash,
+			BlockHash:        r.BlockHash,
+			BlockNumber:      r.BlockNumber,
+			TransactionIndex: r.TransactionIndex,
+			Receipt:          r.Receipt,
+			CreatedAt:        r.CreatedAt,
+		})
+	}
+	return
+}
+
+func fromDBReceiptsPlus(rs []dbReceiptPlus) (receipts []txmgrtypes.ReceiptPlus[evmtypes.Receipt]) {
+	for i := 0; i < len(rs); i++ {
+		r := rs[i]
+		receipts = append(receipts, txmgrtypes.ReceiptPlus[evmtypes.Receipt]{
+			ID:           r.ID,
+			Receipt:      r.Receipt,
+			FailOnRevert: r.FailOnRevert,
+		})
```

Comment:
Lots of allocating and copying here on a relatively hot path. Is there no way to scan directly into the struct you actually want rather than having this intermediary copy?

Reply:
We can do a best-effort and avoid copying fields that have 1:1 mapping between DB type, vs in-memory type. But guaranteeing it may be unavoidable once we support multiple chain families.
The Txmgr will use general interfaces to represent Receipts, Txes, Tx_attempts. However, when these values are serialized into DB, the type may have to change.

Can we use this strategy:
In the db mirroring structs, use pointers to types, not direct types. So when we create the db structs, its just copying references?
So here's how dbReceipt will look like:
type dbReceipt struct {
	ID               *int64
	TxHash           *common.Hash
	BlockHash        *common.Hash
	BlockNumber      *int64
	TransactionIndex *uint
	Receipt         * evmtypes.Receipt
	CreatedAt       * time.Time
}

Reply:
> The Txmgr will use general interfaces to represent Receipts, Txes, Tx_attempts. However, when these values are serialized into DB, the type may have to change.

Why does these have to be separate? Couldn't the types implementing those interfaces just have struct tags or implement methods to serialize/deserialize?

Reply:
If the types can use same struct tags as dbStruct, then we should definitely do that. But we cannot always guarantee.
There might be cases where we encode to the DB differently than how we use in-memory.
In those cases, those fields will have to be serialized/deserialized when being read from or written to the DB.

For example, we may want to convert chainId to a universally unique string for storing in DB. But when using in-memory, each chain family wants to read it as a native chain-id type.

Reply:
But why does that necessitate separate types? Why can't one type handle both responsibilities, like by implementing whatever methods we choose for (de)serialization?

Reply:
There are also options for overriding (de)serialization behavior on a per-field basis, without affecting the in-memory type at all.

Reply:
> We'd be overengineering a solution to require an interface which requires both representations.

Why would it be necessary to require multiple representations?
It's normal and idiomatic in Go to utilize type checks for interface compatibility. We can easily have optional methods for minimal implemenations.